File: androidApp/src/main/kotlin/com/igorwojda/showcase/ui/common/ErrorContent.kt
package com.igorwojda.showcase.ui.common

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun ErrorContent(
    message: String,
    onRetry: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier.padding(all = 24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(text = "⚠️", fontSize = 40.sp)
        Text(
            text = message,
            color = MaterialTheme.colorScheme.error,
            textAlign = TextAlign.Center,
        )
        Button(onClick = onRetry) {
            Text("Reload")
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun ErrorContentPreview() {
    MaterialTheme {
        ErrorContent(message = "Unable to reach the weather service", onRetry = {})
    }
}

File: feature/forecast/src/iosMain/swift/presentation/common/ErrorContent.swift
import SwiftUI

struct ErrorContent: View {
    let message: String
    let onRetry: () -> Void

    var body: some View {
        VStack(spacing: 8) {
            Text("⚠️")
                .font(.system(size: 40))
            Text(message)
                .foregroundStyle(.red)
                .multilineTextAlignment(.center)
            Button("Reload", action: onRetry)
                .buttonStyle(.borderedProminent)
        }
        .padding(24)
    }
}

#Preview {
    ErrorContent(message: "Unable to reach the weather service", onRetry: {})
}
